src/Phinx/Db/Plan/Solver/ActionSplitter.php

```diff
@@ -68,48 +68,45 @@ public function __construct($conflictClass, $conflictClassDual, callable $confli
      * @return \Phinx\Db\Plan\AlterTable[] A list of AlterTable that can be executed without
      * this type of conflict
      */
-    public function __invoke(AlterTable $alter)
+    public function split(AlterTable $alter)
     {
-        $actions = collection($alter->getActions());
-        $conflictActions = $actions
-            ->filter(function ($action) {
-                return $action instanceof $this->conflictClass;
-            })
-            ->toList();
+        $conflictActions = array_filter($alter->getActions(), function ($action) {
+            return $action instanceof $this->conflictClass;
+        });
 
         $originalAlter = new AlterTable($alter->getTable());
         $newAlter = new AlterTable($alter->getTable());
 
-        $actions
-            ->map(function ($action) use ($conflictActions) {
-                if (!$action instanceof $this->conflictClassDual) {
-                    return [$action, null];
-                }
+        $actions = array_map(function ($action) use ($conflictActions) {
+            if (!$action instanceof $this->conflictClassDual) {
+                return [$action, null];
+            }
 
-                $found = false;
-                $matches = $this->conflictFilter;
-                foreach ($conflictActions as $ca) {
-                    if ($matches($ca, $action)) {
-                        $found = true;
-                        break;
-                    }
+            $found = false;
+            $matches = $this->conflictFilter;
+            foreach ($conflictActions as $ca) {
+                if ($matches($ca, $action)) {
+                    $found = true;
+                    break;
                 }
+            }
 
-                if ($found) {
-                    return [null, $action];
-                }
+            if ($found) {
+                return [null, $action];
+            }
 
-                return [$action, null];
-            })
-            ->each(function ($pair) use ($originalAlter, $newAlter) {
-                list($original, $new) = $pair;
-                if ($original) {
-                    $originalAlter->addAction($original);
-                }
-                if ($new) {
-                    $newAlter->addAction($new);
-                }
-            });
+            return [$action, null];
+        }, $alter->getActions());
+
+        foreach ($actions as $pair) {
+            [$original, $new] = $pair;
```

Comment:
This section should be re-written into the above array_map like the others or this leave this as a collection which reads easier.

Reply:
Worked this into the above array_map, which was rewritten to just be a foreach loop now as well.